velox/type/Type.h

```diff
@@ -2008,6 +2008,22 @@ bool registerCustomType(
     const std::string& name,
     std::unique_ptr<const CustomTypeFactories> factories);
 
+std::unordered_map<std::string, std::type_index>& getTypeIndexByOpaqueAlias();
```

Comment:
can you add a bit more documentation here about what this does, when it should be used, and some of the restrictions/considerations you added to the PR summary?